configure.ac

```diff
@@ -2603,6 +2603,13 @@ AS_VAR_IF([ac_cv_gcc_compat], [yes], [
         esac
         AC_MSG_RESULT([$CC])
 
+        # Error on unguarded use of new symbols, which will fail at runtime for
+        # users on older versions of macOS
+        AX_CHECK_COMPILE_FLAG([-Wunguarded-availability-new],
```

Comment:
As noted elsewhere, I think we should use `unguarded-availability` here, not `-new` but I want to test this on older systems which I won't be able to do for several more days.

Reply:
Done in https://github.com/python/cpython/pull/128155/commits/2ddda3a36dff171170a7b7adfa5b5264fe05a4d3. This did not change the output on my machine.

Reply:
(or in CI)